test/format/hs.c

```diff
@@ -64,53 +64,29 @@ hs_cursor(void *arg)
 
     memset(&hs_key, 0, sizeof(hs_key));
     memset(&hs_value, 0, sizeof(hs_value));
-    memset(&key, 0, sizeof(key));
-    hs_start_ts = 0; /* [-Wconditional-uninitialized] */
-    hs_counter = 0;  /* [-Wconditional-uninitialized] */
-    hs_btree_id = 0; /* [-Wconditional-uninitialized] */
-    for (restart = true;;) {
+    for (;;) {
+        /* Open a HS cursor. */
         testutil_check(__wt_curhs_open((WT_SESSION_IMPL *)session, NULL, &cursor));
         F_SET(cursor, WT_CURSTD_HS_READ_COMMITTED);
 
-        /* Search to the last-known location. */
-        if (!restart) {
-            cursor->set_key(cursor, 4, hs_btree_id, &key, hs_start_ts, hs_counter);
-
-            /*
-             * Limit expected errors because this is a diagnostic check (the WiredTiger API allows
-             * prepare-conflict, but that would be unexpected from the history store file).
-             */
-            ret = cursor->search_near(cursor, &exact);
-            testutil_assert(ret == 0 || ret == WT_NOTFOUND || ret == WT_ROLLBACK);
-        }
-
         /*
-         * Get some more key/value pairs. Always retrieve at least one key, that ensures we have a
-         * valid key when we copy it to start the next run.
+         * Move the cursor through the table from the beginning or the end. We can't position the
+         * cursor in the HS store because the semantics of search aren't quite the same as other
+         * tables, and we can't correct for them in application code. We don't sleep with an open
+         * cursor, so we should be able to traverse large chunks of the HS store quickly, without
+         * blocking normal operations.
          */
         next = mmrand(NULL, 0, 1) == 1;
-        for (i = mmrand(NULL, 1, 1000); i > 0; --i) {
-            if ((ret = (next ? cursor->next(cursor) : cursor->prev(cursor))) == 0) {
-                testutil_check(
-                  cursor->get_key(cursor, &hs_btree_id, &hs_key, &hs_start_ts, &hs_counter));
-                testutil_check(cursor->get_value(
-                  cursor, &hs_stop_durable_ts, &hs_durable_timestamp, &hs_upd_type, &hs_value));
-                continue;
+        for (i = mmrand(NULL, 1, 100000); i > 0; ++i) {
```

Comment:
++i？Should be --i?

Reply:
Thanks! Typo of course -- fixed in cbd800b88.